### ISSUE TITLE
self-alignment-X of easy-custom-dynamics

### ISSUE THREAD

**Issue by skiasaurus** [openlilylib/snippets#18]
Usually when I create custom dynamics like `\dynamic "sub. p"`, I have to `\override DynamicText.self-alignment-X = #-1` to make it look better. My problem is that it always has to be a `\once \override` because the default dynamics (e.g. `\p`) look fine the way they are. That becomes a lot to type, and Janek's stated aim in writing the function was to reduce typing. Is there a way to include this in the function?

**Comment by jan-warchol** [comment]
Yes, i know this problem, and i intend to solve it.  But i don't know how long will it take - the "ultimate" solution would be to generalize self-alignment interface, and use the possibilites it will provide.  I have a patch that does this, but it's frozen (i.e. i'm waiting until 2.18 is out, because it cannot be merged before that).

There are two things that might help with getting this done:
- sponsoring (but at this moment i cannot guarantee that it will result in having this issue fixed soon),
- testing and discussing my patch - i'd like to know whether it's working correctly and the interface is good.

Would you like to help with any of these?

**Comment by skiasaurus** [comment]
I can definitely do the latter. :P

**Comment by jan-warchol** [comment]
Great!  What operating system are you on?  Can you compile LilyPond from source? (sorry if i've already asked these questions, but so many things happen that i don't remember...).

**Comment by skiasaurus** [comment]
I’m on Windows 7 x86, and no I can’t. I’m just a lowly end user. :-)

**Comment by jan-warchol** [comment]
Hmm. Unless i learn how to use GUB to provide special binaries myself (lot of work), you cannot test a c++ patch like mine without compiling lilypond from source.

However, i have a script that makes building lilypond quite easy - the only thing you'd have to do is to install Linux, probably in a virtual machine.  It's quite easy, and there are instructions here http://www.lilypond.org/doc/v2.17/Documentation/contributor/lilydev (installation process will probably take ~2hrs)

Do you think you can handle this?

**Comment by skiasaurus** [comment]
Perhaps, but not for about 2 weeks.

**Comment by skiasaurus** [comment]
Well, it looks like it will take significantly longer before I can help out with this. According to my dad/system administrator, my computer (a Dell Latitude D830, vintage 2007) can't handle a virtual machine. I'm saving up to replace it, but barring some unexpected windfall I don't expect it to happen until this summer at the earliest. :-/

**Comment by uliska** [comment]
And what about a Dual Boot which expects significantly less processing power?

**Comment by skiasaurus** [comment]
No good, but he's going to look into building me a virtual machine on our home server.

**Comment by jan-warchol** [comment]
Bad luck.  It would be nice if the home server would work, but - as far as this issue is concerned - don't try too hard.  Considering that i have very little time for Lilypond, it would be probably a much better use of yours (and mine) resources if you focused on the "quick start tutorial" translation (i.e. http://lists.gnu.org/archive/html/lilypond-user/2013-12/msg01198.html).

thanks,
Janek
